pkg/volume/rbd/rbd.go

```diff
@@ -86,30 +94,40 @@ func (plugin *rbdPlugin) GetAccessModes() []api.PersistentVolumeAccessMode {
 }
 
 func (plugin *rbdPlugin) NewMounter(spec *volume.Spec, pod *api.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
-	secret := ""
+	var secret string
+	var err error
 	source, _ := plugin.getRBDVolumeSource(spec)
 
 	if source.SecretRef != nil {
-		kubeClient := plugin.host.GetKubeClient()
-		if kubeClient == nil {
-			return nil, fmt.Errorf("Cannot get kube client")
-		}
-
-		secretName, err := kubeClient.Core().Secrets(pod.Namespace).Get(source.SecretRef.Name)
-		if err != nil {
+		if secret, err = plugin.getSecret(pod.Namespace, source.SecretRef.Name); err != nil {
 			glog.Errorf("Couldn't get secret %v/%v", pod.Namespace, source.SecretRef)
 			return nil, err
 		}
-		for name, data := range secretName.Data {
-			secret = string(data)
-			glog.V(1).Infof("ceph secret info: %s/%s", name, secret)
-		}
-
 	}
+
 	// Inject real implementations here, test through the internal function.
 	return plugin.newMounterInternal(spec, pod.UID, &RBDUtil{}, plugin.host.GetMounter(), secret)
 }
 
+func (plugin *rbdPlugin) getSecret(namespace, secretName string) (string, error) {
```

Comment:
I like the idea to move this into `volume/util.go` but could we change it from: 
- `func getSecret(namespace, secretName string) (string, error)`
- `func getSecret(namespace, secretName string) (map[string]string, error)`

Since admins can save multiple `data/secrets` (I don't know how to correctly name it) in a Kubernetes secret we should parse all data in this secret like this https://github.com/johscheuer/kubernetes/commit/290480ac1d1736395e4f52cd546ab6f75df6ba66#diff-43774ec582be79d3c984b30f79f89f64R140 where `name` is the key for the map and `data` is the value.